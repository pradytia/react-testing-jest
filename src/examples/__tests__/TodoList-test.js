import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../TodoList';

const todos = [
  'Buy apple',
  'Buy banana',
  'Do laundry',
];

describe('TodoList component', () => {
  it('should render with no list', () => {
    let component = renderer.create(<TodoList />);
    expect(component).toMatchSnapshot();
  });

  it('should render with list', () => {
    let component = renderer.create(<TodoList todos={todos} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with filtered list', () => {
    let keywords = ['', 'b', 'buy a'];
    keywords.forEach((keyword) => {
      const component = renderer.create(<TodoList todos={todos} keyword={keyword} />);
      expect(component).toMatchSnapshot();
    })
  });
});
