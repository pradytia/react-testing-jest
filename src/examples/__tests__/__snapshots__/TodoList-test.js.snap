// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should render with filtered list 1`] = `
<div>
  <ul
    style={
      Object {
        "margin": 0,
        "padding": 0,
        "width": 300,
      }
    }
  >
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Buy apple
    </li>
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Buy banana
    </li>
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Do laundry
    </li>
  </ul>
</div>
`;

exports[`TodoList component should render with filtered list 2`] = `
<div>
  <ul
    style={
      Object {
        "margin": 0,
        "padding": 0,
        "width": 300,
      }
    }
  >
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Buy apple
    </li>
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Buy banana
    </li>
  </ul>
</div>
`;

exports[`TodoList component should render with filtered list 3`] = `
<div>
  <ul
    style={
      Object {
        "margin": 0,
        "padding": 0,
        "width": 300,
      }
    }
  >
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Buy apple
    </li>
  </ul>
</div>
`;

exports[`TodoList component should render with list 1`] = `
<div>
  <ul
    style={
      Object {
        "margin": 0,
        "padding": 0,
        "width": 300,
      }
    }
  >
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Buy apple
    </li>
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Buy banana
    </li>
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      Do laundry
    </li>
  </ul>
</div>
`;

exports[`TodoList component should render with no list 1`] = `
<div>
  <ul
    style={
      Object {
        "margin": 0,
        "padding": 0,
        "width": 300,
      }
    }
  >
    <li
      style={
        Object {
          "borderBottom": "1px solid #999",
          "fontSize": 16,
          "listStyleType": "none",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      No list
    </li>
  </ul>
</div>
`;
